### PR TITLE
WIP feat: double tap titles and descriptions to edit

### PR DESCRIPTION
**Pull request by williamconnolly**
As a user, I kept finding myself double-tapping titles and descriptions to edit them. I think it comes from me being used to a file explorer's "double click file to rename it" functionality. 

Also, on my tablet (and desktop screen as well), the body of the application was too narrow. Rather than 35em, I changed to 80% and found it was more usable. I'm open to removing this of course, if others' opinions differ.

### CODE REVIEW

File: bin/compile_search_parser.sh [hunk 1]
```diff
@@ -6,5 +6,10 @@ npx pegjs -o src/lib/headline_filter_parser.{js,grammar.pegjs}
 # code that we're writing and we want to keep eslint warnings at 0,
 # we're telling eslint to ignore the generated file.
 PARSER_FILE=src/lib/headline_filter_parser.js
+TMP_FILE=/tmp/headline_filter_parser.js
+
 # Prepend the generated parser code with a `eslint-disable` comment.
-sed -i '1i \/* eslint-disable */' "$PARSER_FILE"
+echo '/* eslint-disable */' > "$TMP_FILE"
```
Comment (williamconnolly) [inline bin/compile_search_parser.sh]: This sed command failed to execute on my computer. I have had issues with sed incompatibilities with linux/macOS etc, so for this specific action, "pre-pend a line to the file", I got it working this way. I'm specifically on macOS Mojave  10.14.15. I open to reverting this change as well.

Comment (schoettl) [reply bin/compile_search_parser.sh]: Can you tell me the output of `sed --version`? Maybe you can just install a modern GNU sed on your system?

I'd generally prefer this sed one-liner. If it have to be using a temp file, it should be done with `mktemp`, which is also hard to make platform-independent. But a static temp file can randomly make problems to some users, e.g. when they have a local CI system running.

**Comment by schoettl** [comment]
I tested it, but double clicks to edit doesn't seem to work. I tried in chromium, normal mode and mobile emulation mode.

What is the behavior for a normal click, now? Still collapse/expand header? Isn't this delayed, when double clicks are also handled?

Having a width of 80 % makes the toolbar look not good. But it may indeed be more usable. And if we change the UI concept anyway (material design), it's maybe OK.

**Comment by munen** [comment]
I tested it, too. The feature does work for me (after enabling it in the settings) - tested on FF (Linux) and iOS.

However, I'm not convinced of the UX of this feature atm. Since clicking a header will toggle an expand/collapse, a double click will almost always miss it's mark. I've created a demo with three headlines. I'm always double clicking, but it almost always doesn't work, because of expanding/collapsing headlines.

demo

Apart from that, on a mobile device, it's easy to trigger any action from the `HeaderActionBar` on a 'double-tap'. The first tap will open the header up and the second will click anything.

As of this state, I'm afraid we cannot merge it. @williamconnolly if you can address these UX features, this assassment can change, of course. Personally, I wouldn't know how to solve it, though. It's hard to do two things with one action.

Apart from the UX, the code is nicely done. Good job on building your first neat PR! :pray: 

I'm very sorry I don't have better news on the review, though.

**Comment by williamconnolly** [comment]
Sounds good, I'll look into fixing this. I started compensating for the quick-header-collapse behavior and got consistent title double-taps when the titles were scrolled high enough in the body - I don't yet have a solution for this. All online tutorials speak of debouncing the taps for collapsing, to give room for onDoubleTap to fire. Thanks for feedback. What would you think instead of Tap-And-Hold?

File: bin/compile_search_parser.sh (hunk 1 above)
Comment (williamconnolly) [reply bin/compile_search_parser.sh]: Ahh, I see, the local CI system that may not have a tmp dir is a good argument. I'll futz around with `brew install gsed` or something - I know that macOS default unix utils are woefully out of date.

**Comment by munen** [comment]
"Tap-And-Hold" sounds good for editing the title.

Having said that, at some point "Tap-And-Hold" might trigger a new context menu which opens non-primary actions for headers. There has been previous discussion going on around this. The rationale is that the current `HeaderActionBar` is already filled with actions and it'll be an issue adding more to that. The whole paradigm around how to trigger actions against headlines may change in the future. There's nothing decided, though. This is an issue where some open discussion has happened: https://github.com/200ok-ch/organice/issues/188

File: bin/compile_search_parser.sh (hunk 1 above)
Comment (munen) [reply bin/compile_search_parser.sh]: :+1:

**Comment by williamconnolly** [comment]
Closing for now as I dont have much time to complete due to obvious circumstances

**Comment by munen** [comment]
All the best for your other commitments! :pray:

**Pull request closed without merging**